src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/DoNotCatchGeneralExceptionTypesTests.cs

```diff
@@ -0,0 +1,571 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.ApiDesignGuidelines.Analyzers.UnitTests
+{
+    public class DoNotCatchGeneralExceptionTypesTests : DiagnosticAnalyzerTestBase
+    {
+        protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
+        {
+            return new DoNotCatchGeneralExceptionTypesAnalyzer();
+        }
+
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new DoNotCatchGeneralExceptionTypesAnalyzer();
+        }
+
+        [Fact]
+        public void CSharp_Diagnostic_GeneralCatch()
+        {
+            VerifyCSharp(@"
+            using System;
+            using System.IO;
+
+            namespace TestNamespace
+            {
+                class TestClass
+                {
+                    public static void TestMethod()
+                    {
+                        try 
+                        {
+                            FileStream fileStream = new FileStream(""name"", FileMode.Create);
+                        }
+                        catch (IOException e)
+                        {
+                        }
+                        catch
+                        {
+                        }
+                    }
+                }
+            }",
+            GetCA1031CSharpResultAt(18, 25, "TestNamespace.TestClass.TestMethod()"));
```

Comment:
We likely want to use just the method name here - in past attempting to use display string has caused diagnostic messages to blow up quite a bit. It is fine to file a tracking issue for that change though.

Reply:
@m-akinc Let me know your preference - fixing here or filing a tracking issue. I can merge the PR if latter is preferable.

Reply:
I'll fix it now